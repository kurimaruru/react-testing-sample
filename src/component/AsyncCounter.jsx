import { useState } from "react"

export function AsyncCounter() {
  const [count, setCount] = useState(0)
  const [isLoading, setLoading] = useState(false)
  const handler = () => {
    setLoading(true)
    setTimeout(() => {
      setCount(count + 1)
      setLoading(false)
    }, 1000)
  }
  return (
    <>
      <div>AsyncCount: {count}</div>
      <div>
        {isLoading && <span>...Loading</span>}
        <button onClick={handler} disabled={isLoading}>
          AsyncIncrement
        </button>
      </div>
    </>
  )
}
