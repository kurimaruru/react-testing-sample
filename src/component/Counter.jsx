import { useState } from "react";

export const Counter = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <h1>Count:{count}</h1>
      <div>
        <button onClick={() => setCount(count + 1)}>increment</button>
      </div>
    </>
  );
};
