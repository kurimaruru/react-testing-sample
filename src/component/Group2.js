export const getBeverage = age =>{
  const filteredBeverages = () =>{
    let drink = age >= 20 ?'beer':'cola';
    return drink;
  }

  const beverage = filteredBeverages();
  return beverage;
}
