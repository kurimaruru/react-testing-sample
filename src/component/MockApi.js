import axios from "axios";

class Users {
  static all(){
    const name = {users:[{name:"Bob"}]};
    return axios.get('/users.json').then(() =>{return name;});
  }
}

export default Users;
