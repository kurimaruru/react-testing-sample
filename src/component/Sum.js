// const sum = (a, b) => a + b;
// module.exports = sum;

export const sum = (a,b) => a+b ;
