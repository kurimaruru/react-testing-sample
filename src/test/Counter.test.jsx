import {cleanup,fireEvent,render,screen} from "@testing-library/react";
import {Counter} from "../component/Counter";
//Group of Counter test
describe("Counter",() =>{
  //when test finish,do to clean up
  afterEach(() =>{
    cleanup();
  })
  //snap shot test
  test("render",() =>{
    const {asFragment} = render(<Counter/>);
    expect(asFragment()).toMatchSnapshot();
  });

  test("click:count" ,() =>{
    render(<Counter/>);
    const button = screen.getByText("increment");
    //user operation
    fireEvent.click(button);
    fireEvent.click(button);
    screen.getByText("Count:2");
  });
})
