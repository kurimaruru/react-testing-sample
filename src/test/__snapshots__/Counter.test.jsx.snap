// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render 1`] = `
<DocumentFragment>
  <h1>
    Count:0
  </h1>
  <div>
    <button>
      increment
    </button>
  </div>
</DocumentFragment>
`;
